user/troubleshooting/pci-troubleshooting.md

```diff
@@ -0,0 +1,147 @@
+---
+layout: doc
+title: PCI Troubleshooting
+permalink: /doc/pci-troubleshooting/
+---
+
+# PCI troubleshooting #
+
+## DMA errors ##
+
+VMs with attached PCI devices in Qubes have allocated a small buffer for DMA operations (called swiotlb).
+By default, it is 2MB, but some devices (such as the [Realtek RTL8111DL Gigabit Ethernet Controller](https://groups.google.com/group/qubes-devel/browse_thread/thread/631c4a3a9d1186e3)) need a larger DMA buffer size. 
+Without a larger buffer, you will face DMA errors such as `Failed to map TX DMA`.
+
+To change this allocation, edit VM's kernel parameters (this is expressed in 512B chunks) by running the following in a dom0 terminal:
+
+    # qvm-prefs netvm |grep kernelopts
+    kernelopts       : iommu=soft swiotlb=2048 (default)
+    # qvm-prefs -s netvm kernelopts "iommu=soft swiotlb=8192"
```

Comment:
While this may be needed for some devices in some form, the 8192 value is the default already (some devices may require larger one).